src/Simplex/Chat.hs

```diff
@@ -835,7 +835,26 @@ processChatCommand = \case
             let aciContent = ACIContent SMDRcv $ CIRcvGroupInvitation ciGroupInv {status = CIGISAccepted} memRole
             updateDirectChatItemView userId ct itemId aciContent Nothing
           _ -> pure () -- prohibited
-  APIMemberRole _groupId _groupMemberId _memRole -> throwChatError $ CECommandError "unsupported"
+  APIMemberRole groupId memberId memRole -> withUser $ \user -> do
+    Group gInfo@GroupInfo {membership} members <- withStore $ \db -> getGroup db user groupId
+    if memberId == groupMemberId' membership
+      then changeMemberRole user gInfo members membership $ SGEUserRole memRole
+      else case find ((== memberId) . groupMemberId') members of
+        Nothing -> throwChatError CEGroupMemberNotFound
+        Just m -> changeMemberRole user gInfo members m $ SGEMemberRole memberId memRole
+    where
+      changeMemberRole user gInfo@GroupInfo {membership} members m@GroupMember {memberId = mId, memberRole = mRole, memberStatus = mStatus} gEvent = do
+        let userRole = memberRole (membership :: GroupMember)
+            canChangeRole = userRole >= GRAdmin && userRole >= mRole && userRole >= memRole && memberCurrent membership
```

Comment:
I'm not sure it's correct we allow for role to govern same level role.. For remove as well - e.g. for owners to remove/change other owners role. As a bonus it would allow less role changes - less possibilities for fragmentation at the current stage

Reply:
Thought about it too - I think it's ok for admins to be able to downgrade other admins...